src/ray/raylet/scheduling/local_task_manager.cc

```diff
@@ -346,10 +347,12 @@ void LocalTaskManager::SpillWaitingTasks() {
 
 bool LocalTaskManager::TrySpillback(const std::shared_ptr<internal::Work> &work,
                                     bool &is_infeasible) {
-  std::string node_id_string =
-      GetBestSchedulableNode(*work,
-                             /*spill_waiting_task=*/false,
-                             /*force_spillback=*/false, &is_infeasible);
+  std::string node_id_string = cluster_resource_scheduler_->GetBestSchedulableNode(
+      work->task.GetTaskSpecification(),
+      /*prioritize_local_node*/
+      (work->grant_or_reject || work->is_selected_based_on_locality),
```

Comment:
overall lgtm. The only concern is that this logic is repeated three times, should we create a helper function like work->prioritize_local_node()?

Reply:
yeah... i'm a bit confused by the behavior of `grant_or_reject`.

actually if `grant_or_reject` is true, should we bypass the distributed scheduling part and only do local scheduling?